hpotter/plugins/ContainerThread.py

```diff
@@ -41,8 +41,9 @@ def connect_to_container(self):
         for x in range(9):
             try:
                 self.dest = socket.create_connection( \
-                    (self.container_ip, self.container_port), timeout=2)
-                self.dest.settimeout(None)
+                    (self.container_ip, self.container_port))
+                timeout = self.config.get('container_socket_timeout')
+                self.dest.settimeout(timeout)
```

Comment:
set timeout on connection to docker container so doesn't hang in receive method. This allows control to return to ContainerThread eventually so the container can be shut down.